mobile/src/main/java/org/openhab/habdroid/core/OnUpdateBroadcastReceiver.java

```diff
@@ -0,0 +1,40 @@
+package org.openhab.habdroid.core;
+
+import android.content.BroadcastReceiver;
+import android.content.Context;
+import android.content.Intent;
+import android.content.SharedPreferences;
+import android.preference.PreferenceManager;
+import android.util.Log;
+
+import org.openhab.habdroid.util.Constants;
+
+import static org.openhab.habdroid.util.Constants.PREFERENCE_COMPAREABLEVERSION;
+
+public class OnUpdateBroadcastReceiver extends BroadcastReceiver {
+    private static final String TAG = OnUpdateBroadcastReceiver.class.getSimpleName();
+
+    private static final int UPDATE_LOCAL_CREDENTIALS = 25;
+
+    @Override
+    public void onReceive(Context context, Intent intent) {
+        if (!Intent.ACTION_MY_PACKAGE_REPLACED.equalsIgnoreCase(intent.getAction())) {
+            return;
+        }
+        SharedPreferences prefs = PreferenceManager.getDefaultSharedPreferences(context);
+
+        if (prefs.getInt(PREFERENCE_COMPAREABLEVERSION, 0) <= UPDATE_LOCAL_CREDENTIALS) {
+            Log.d(TAG, "Checking for putting username/password to local username/password.");
+            SharedPreferences.Editor edit = prefs.edit();
+            if (prefs.getString(Constants.PREFERENCE_LOCAL_USERNAME, null) == null) {
+                edit.putString(Constants.PREFERENCE_LOCAL_USERNAME, prefs.getString(Constants
+                        .PREFERENCE_USERNAME, null));
+            }
+            if (prefs.getString(Constants.PREFERENCE_LOCAL_PASSWORD, null) == null) {
+                edit.putString(Constants.PREFERENCE_LOCAL_PASSWORD, prefs.getString(Constants
+                        .PREFERENCE_PASSWORD, null));
+            }
+            edit.apply();
```

Comment:
Wouldn't you want to update PREFERENCE_COMPARABLEVERSION to the new version here? I realize the case of the user not starting the activity before the next update might be a bit academic, but fixing this doesn't hurt either I think.

Reply:
You're right, haven't thought about that :) I think it should always put the new version into the setting, even if no migration of data is needed :)